Sources/TuistLoader/Utils/ManifestFilesLocator.swift

```diff
@@ -73,18 +73,40 @@ public final class ManifestFilesLocator: ManifestFilesLocating {
         } else {
             let path = rootDirectoryLocator.locate(from: locatingPath) ?? locatingPath
 
-            let excludedPaths = excluding
-                .flatMap {
-                    FileHandler.shared.glob(path, glob: $0)
+            let pluginsPaths = fetchTuistManifestsFilePaths(at: path)
+                .filter { $0.basename == Manifest.plugin.fileName(path) }
+                .filter { path in
+                    !excluding.contains { pattern in match(path, pattern: pattern) }
                 }
 
-            let pluginsPaths = Set(FileHandler.shared.glob(path, glob: "**/\(Manifest.plugin.fileName(path))"))
-                .subtracting(excludedPaths)
-
             return Array(pluginsPaths)
         }
     }
 
+    private func match(_ path: AbsolutePath, pattern: String) -> Bool {
+        return fnmatch(pattern, path.pathString, 0) != FNM_NOMATCH
+    }
+
+    var cacheTuistManifestsFilePaths = [AbsolutePath: Set<AbsolutePath>]()
+
+    private func fetchTuistManifestsFilePaths(at path: AbsolutePath) -> Set<AbsolutePath> {
+        if let cachedTuistManifestsFilePaths = cacheTuistManifestsFilePaths[path] {
+            return cachedTuistManifestsFilePaths
+        }
+        let fileNamesCandidates: Set<String> = [
+            Manifest.project.fileName(path),
+            Manifest.workspace.fileName(path),
+            Manifest.plugin.fileName(path),
+        ]
+        let tuistManifestSignature = "import ProjectDescription"
+        let tuistManifestsFilePaths = FileHandler.shared.glob(path, glob: "**/*.swift")
+            .filter { fileNamesCandidates.contains($0.basename) }
+            .filter { (try? FileHandler.shared.readTextFile($0).contains(tuistManifestSignature)) ?? false }
```

Comment:
I remember sometime we discussed allowing empty manifests, in case you create an empty one and then invoke tuist edit on it, I think it would make sense to have that logic

Reply:
Good point @danyf90 , done.